pyctcdecode/decoder.py

```diff
@@ -609,6 +665,65 @@ def _decode_logits(
         ]
         return output_beams
 
+    def get_starting_state(self) -> Tuple[List[Beam], LMScoreCache, Dict[str, float]]:
+        """Get the starting beams and initial caches."""
+        start_beam = [EMPTY_START_BEAM]
+        language_model = self._language_model
+        if language_model is None:
+            cached_lm_scores: LMScoreCache = {}
+        else:
+            start_state = language_model.get_start_state()
+            cached_lm_scores = {("", False): (0.0, 0.0, start_state)}
+        cached_p_lm_scores: Dict[str, float] = {}
+        return start_beam, cached_lm_scores, cached_p_lm_scores
+
+    def partial_decode_beams(
```

Comment:
this approach depends on the user for management of state like caches, previous beams, and the hotword sscorer. 

Another approach would be to add `initialize_partial_decoding_state()` and `clear_partial_decoding_state()` functions and preserving these things within the decoder object

A third approach would be to add an `initialize` kwarg to run on the first call and then clear state when `is_end==True`. 

Not sure which of these approaches would be best, they're all more or less equivalent but would be used differently by the user

Reply:
after doing some more thinking, I think this approach is the correct one to go with - this allows the most freedom to the user. if state is held within the decoder, then a decoder object can only process one audio stream at a time. this way allows the decoder to process many streams - the state just has to be preserved somewhere else and called up later. this would also allow for better scaling. for a language model where the state is saved in a way that is safe for multiprocessing, we could run many decoder servers and it wouldn't matter which one a given chunk went to if the state is made accessible to all